### PR TITLE
Add overlapping windowing system

### PR DESCRIPTION
**Pull request by aeschylus**

### CODE REVIEW

File: src/components/WorkspaceFreeform.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        }}
+        onDragStop={(e, d) => {
+          setWorkspaceViewportPosition({ x: d.x, y: d.y });
+        }}
+        cancel={`.${ns('window')}`}
+        className={ns('workspace')}
+      >
+        {
+          Object.values(windows).map(window => (
+            <Window
```
Comment (mejackreed) [inline src/components/WorkspaceFreeform.js]: Could this be a seam to render the `draggableWorkspaceWindow` (perhaps as a component) with a child of `Window`? Could that remove the conditional from `Window`?

Comment (aeschylus) [reply src/components/WorkspaceFreeform.js]: Yeah, that's probably nicer. There was a lot of reshuffling. I'll make this update.

File: package.json [hunk 2]
```diff
@@ -44,13 +45,14 @@
     "node-sass": "^4.9.2",
     "openseadragon": "^2.4.0",
     "prop-types": "^15.6.2",
-    "react": "^16.8.3",
-    "react-dom": "^16.8.3",
+    "react": "^16.7.0",
```
Comment (mejackreed) [inline package.json]: Do we need to specifically use 16.7 ? or is this just maybe rebase cruft?

File: package.json [hunk 3]
```diff
@@ -31,6 +31,7 @@
     "@material-ui/core": "^3.9.1",
     "@material-ui/icons": "^3.0.2",
     "@researchgate/react-intersection-observer": "^0.7.4",
+    "acorn": "6.1.1",
```
Comment (mejackreed) [inline package.json]: Where is acorn being used?

File: package.json [hunk 4]
```diff
@@ -85,6 +87,7 @@
     "jest": "^24.1.0",
     "jest-fetch-mock": "^2.1.1",
     "jest-puppeteer": "^4.0.0",
+    "jsdom": "13.2.0",
```
Comment (mejackreed) [inline package.json]: Should we remove the eslint-inline disable here now?: https://github.com/ProjectMirador/mirador/blob/master/setupJest.js#L3

File: src/containers/Window.js [hunk 5]
```diff
@@ -2,12 +2,13 @@ import { compose } from 'redux';
...
 import { Window } from '../components/Window';
 
+
 /**
  * mapStateToProps - used to hook up connect to action creators
  * @memberof Window
  * @private
  */
-const mapStateToProps = ({ manifests, windows }, props) => ({
+const mapStateToProps = ({ config, manifests, windows }, props) => ({
```
Comment (mejackreed) [inline src/containers/Window.js]: Is config needed here still?

File: __tests__/src/actions/workspace.test.js [hunk 6]
```diff
@@ -49,4 +49,29 @@ describe('workspace actions', () => {
...
+      expect(actions.setWorkspaceViewportPosition({
+        x: 20,
+        y: 20,
+      })).toEqual(expectedAction);
+    });
+  });
+  describe('toggleWorkspaceExposeMode', () => {
+    it('should set the exposeMode to true', () => {
+      const expectedAction = {
+        type: ActionTypes.TOGGLE_WORKSPACE_EXPOSE_MODE,
```
Comment (mejackreed) [inline __tests__/src/actions/workspace.test.js]: Is `TOGGLE_WORKSPACE_EXPOSE_MODE` just a placeholder in the state to enable future work?

File: package.json (hunk 3 above)
Comment (aeschylus) [reply package.json]: It's a peer dependency that was uncovered when I was forced to run things on a new machine.

File: src/styles/index.scss [hunk 7]
```diff
@@ -35,6 +43,8 @@
   }
 
   &-window {
+    width:100%;
+    height:100%;
```
Comment (mejackreed) [inline src/styles/index.scss]: Is this perhaps a duplicate?

File: src/styles/index.scss [hunk 8]
```diff
@@ -1,21 +1,29 @@
 @import 'variables';
 
 .mirador {
-  &-app {
-    height: 100%;
-    position: relative;
-    width: 100%;
+  &-viewer {
+    position: absolute;
```
Comment (mejackreed) [inline src/styles/index.scss]: I'm seeing some style warnings in our `npm run lint` task. Should we fix those now or spin off a ticket for that?

File: package.json (hunk 3 above)
Comment (mejackreed) [reply package.json]: I'm not sure i understand that. Wouldn't the package list that as a dependency if it was needed? Why wouldn't our CI/CD catch that?

**Comment by mejackreed** [comment]
@aeschylus not sure if this is a big fix or not, but I also just noticed that now our multiple "embedded" versions are broken: https://deploy-preview-1815--mirador-dev.netlify.com/__tests__/integration/mirador/embedding.html

**Comment by aeschylus** [comment]
The styling on the integration tests is unrelated. The mirador viewer is meant to fill the bounds of its parent. Therefore the parent will need to have `position:relative` or `position:absolute` to provide the coordinate system. `overflow:hidden` should probably also be required, but that can be addressed elsewhere. In general there are more considerations around how we want to treat the configured parent box.

File: src/styles/index.scss (hunk 7 above)
Comment (mejackreed) [reply src/styles/index.scss]: I think `height` was specified twice in the css. It seems resolved now.

**Comment by mejackreed** [comment]
I created a spin off ticket to fix this: https://github.com/ProjectMirador/mirador/issues/2007

**Comment by mejackreed** [comment]
Adding this to the PR so its not lost in comments on Slack:
 ### To switch workspace type
```javascript
miradorInstance.store.dispatch(miradorInstance.actions.updateConfig({
  workspace: {
    type: "elastic"
  }
}))
```

**Pull request merged**